src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -801,6 +801,9 @@ public void Disconnect(string reason, Exception exception = null)
 
             if (this.IsConnected) this.SetStateAsync(NetworkPeerState.Disconnecting).GetAwaiter().GetResult();
 
+            this.Behaviors?.ToList().ForEach(b => b?.Detach());
```

Comment:
Why? No need to detach here.

Reply:
`ToList()` should also use null propagation